R-package/R/aliases.R

```diff
@@ -6,6 +6,7 @@
 #               R sources files during installation).
 # [return] A named list, where each key is a parameter relevant to lgb.DataSet and each value is a character
 #          vector of corresponding aliases.
+,,,,
```

Comment:
I'm starting by intentionally pushing a change that raises an ERROR in `R CMD check`, so we can see that the Windows tests pass.

Reply:
even with this syntax error, all of the Windows tests passed!

https://github.com/microsoft/LightGBM/pull/3435/checks?check_run_id=1204672524

![image](https://user-images.githubusercontent.com/7608904/95008629-3632a000-05e1-11eb-92c7-29229a062fe0.png)

![image](https://user-images.githubusercontent.com/7608904/95008643-46e31600-05e1-11eb-8eec-1d6acd7391b8.png)

Reply:
After adding https://github.com/microsoft/LightGBM/pull/3435/commits/f9165756bbae605f7710058dde878cd325da4acb, the MINGW tests fail as expected

https://github.com/microsoft/LightGBM/pull/3435/checks?check_run_id=1204732555

![image](https://user-images.githubusercontent.com/7608904/95008894-49df0600-05e3-11eb-8527-7baa0066d9e8.png)

But the MSVC ones do not! Another issue with how errors are caught by `Run-R-Code-Redirect-Stderr`, maybe.

Reply:
after adding https://github.com/microsoft/LightGBM/pull/3435/commits/e1916737de3d4d71e456da70a83b69af89c7e6d1, all Windows checks now fail as expected

https://github.com/microsoft/LightGBM/pull/3435/checks?check_run_id=1204756009

![image](https://user-images.githubusercontent.com/7608904/95009173-93c8eb80-05e5-11eb-8f2a-bb630d113428.png)